src/main/java/org/broadinstitute/hellbender/utils/SimpleInterval.java

```diff
@@ -36,6 +41,70 @@ public SimpleInterval(final String contig, final int start, final int end){
         this.end = end;
     }
 
+    /**
+     * Makes an interval by parsing the string. Semantics of start and end are defined in {@link Locatable}.
```

Comment:
This is much improved, but I still think we need a very prominent annotated `@warning` near the top: "does not fill in the true contig end values for intervals that reach to the end of their contig, uses `Integer.MAX_VALUE` instead" (this bug-inducing bit of information is currently buried inside the table of formats below).